### PR TITLE
Report carve dir

### PR DESCRIPTION
**Pull request by e3krisztian**
Reworked #891 to report carve dirs instead of the carved files, as well as support different suffixes for carve and extraction directories.

### CODE REVIEW

File: unblob/processing.py
Comment (qkaiser) [inline unblob/processing.py:105]: function name does not say a lot about its purpose. How about `get_path_under_root` ?

File: unblob/processing.py
Comment (qkaiser) [inline unblob/processing.py:119]: Should be something like:

```py
    def get_extract_dir_for(self, path: Path) -> Path:
        return self.intern(path.with_name(path.name + self.extract_suffix))

    def get_carve_dir_for(self, path: Path) -> Path:
        return self.intern(path.with_name(path.name + self.carve_suffix))
```

File: unblob/cli.py
Comment (qkaiser) [inline unblob/cli.py:268]: What does this means ?

File: unblob/processing.py
Comment (qkaiser) [inline unblob/processing.py:555]: That is true only if the user set the `-k` option or the `--keep-extracted-chunks`. No ?

File: unblob/processing.py
Comment (qkaiser) [inline unblob/processing.py:618]: missing type hint

File: unblob/processing.py
Comment (qkaiser) [reply unblob/processing.py:555]: It is actually confusing because in `_extract_chunk` you do:

```py
if remove_extracted_input and not self.config.keep_extracted_chunks:
```

File: unblob/cli.py
Comment (qkaiser) [inline unblob/cli.py:496]: This whole logic change is unrelated to the display of output path. Should be placed in a separate commit.

File: unblob/cli.py
Comment (qkaiser) [inline unblob/cli.py:471]: I would put the output path in the summary panel, first line.

**Review by qkaiser — COMMENTED** [review]
Nice work. I would maybe add some tests where we set the `ExtractionConfig` to use non-standard `carve_suffix` and `extract_suffix` ? See if we can spot abnormal behavior that would appear out of assumption made on the default suffix values.

**Comment by qkaiser** [comment]
Commits f2cb06ef36501c447b8ee7161cb7f1b47852dc05 and 8024aede679f0d1b13d253826edf8635a20fd48e should be next to each other in the history I think. Easier to follow.

File: unblob/cli.py
Comment (e3krisztian) [reply unblob/cli.py:268]: It means, that the rest of the arguments are keyword only arguments: must be supplied with a name.

The problem with parameters of generic types is, that at the call site it can be confusing, and not easily understandable without looking at the function signature.  Boolean and optional parameters are the most confusing.

`f(True, False, None, True)` would hard to understand without knowing the parameter names, which hopefully are self-describing.

File: unblob/processing.py
Comment (e3krisztian) [reply unblob/processing.py:618]: added.

File: unblob/processing.py
Comment (e3krisztian) [reply unblob/processing.py:555]: > That is true only if the user set the `-k` option or the `--keep-extracted-chunks`. No ?

No, we keep the carved chunks in this case, as here the file is from the extraction (*whole file chunk*), which we do not carve, thus we want to keep input file even if extracted (we want to remove only carved files).

> It is actually confusing because in `_extract_chunk` you do:
> 
> ```python
> if remove_extracted_input and not self.config.keep_extracted_chunks:
> ```

Thanks, this is good catch, moved the condition to the other call site in `_carve_then_extract_chunk`, where it belongs.

File: unblob/processing.py
Comment (e3krisztian) [reply unblob/processing.py:119]: yeah, it was like that, however I wanted to
- say more about the input of `get_extract_dir_for`
- preserve the visual similarity of the logic by not using different variable name

I was not happy with this, so converted as suggested

File: unblob/processing.py
Comment (e3krisztian) [reply unblob/processing.py:105]: Changed to `_get_output_path`.

File: unblob/cli.py
Comment (e3krisztian) [reply unblob/cli.py]: Separated. The logic change is a fix commit now, as in certain failure cases there was no user feedback due to an early return in the function.

Separated as fix-commit, moved `Chunks identification ratio` from `Summary` to after chunk distribution table, as chunk stat was causing the silent failure, this change lets both `Summary` and `Encountered errors` to be printed, while keeping chunk stat in one block.

File: unblob/cli.py
Comment (e3krisztian) [reply unblob/cli.py]: Cool! Done.

**Comment by qkaiser** [comment]
@e3krisztian cool ! I'll wait for the tests to land before I do a final review round.

File: tests/test_processing_suffixes.py
Comment (qkaiser) [inline tests/test_processing_suffixes.py:41]: It feels something's hidden here. How is unblob made to extract into `expected_output_dir_name` ? Shouldn't unblob extract into `chunks_c` or `chunks_carve` as top-level ? What am I missing ?

**Comment by qkaiser** [comment]
@e3krisztian two small comments. I think we can rebase and merge when it's cleared.

File: tests/test_processing_suffixes.py
Comment (e3krisztian) [reply tests/test_processing_suffixes.py]: Yep, I was not happy with this either. Changed the input-output layout and renamed the variable to avoid confusion.

**Comment by e3krisztian** [comment]
Rebased and resolved raised problems in tests.

**Review by qkaiser — APPROVED** [review]

**Pull request merged**
